__tests__/app.test.js

```diff
@@ -112,6 +112,101 @@ describe("/api/articles/:article_id", () => {
                 expect(msg).toBe("invalid article id");
             });
     });
+
+    // PATCH requests
+    test("PATCH - 200: returns updated article when passed an object with a inc_votes property", () => {
+        return request(app)
+            .patch("/api/articles/1")
+            .send({ inc_votes: 20 })
+            .expect(200)
+            .then(({ body: { article } }) => {
+                expect(article).toEqual({
+                    article_id: 1,
+                    title: "Living in the shadow of a great man",
+                    topic: "mitch",
+                    author: "butter_bridge",
+                    body: "I find this existence challenging",
+                    created_at: expect.any(String),
+                    votes: 120
+                });
+            });
+    });
+    test("PATCH - 200: returns as normal, ignoring superfluous properties", () => {
+        return request(app)
+            .patch("/api/articles/1")
+            .send({
+                inc_votes: 20,
+                topic: "hello",
+                actorWhoPlaysDrEvilAustinPowersAndFatBastardInTheRenownedAndExcellentAustinPowersTrilogy:
+                    "Mike Myers"
+            })
+            .expect(200)
+            .then(({ body: { article } }) => {
+                expect(article).toEqual({
+                    article_id: 1,
+                    title: "Living in the shadow of a great man",
+                    topic: "mitch",
+                    author: "butter_bridge",
+                    body: "I find this existence challenging",
+                    created_at: expect.any(String),
+                    votes: 120
+                });
+            });
+    });
+    test("PATCH - 200: sets votes property to 0 if passed object makes it negative", () => {
+        return request(app)
+            .patch("/api/articles/1")
+            .send({ inc_votes: -200 })
+            .expect(200)
+            .then(({ body: { article } }) => {
+                expect(article.votes).toBe(0);
+            });
+    });
+    test("PATCH - 200: rounds inc_votes down to nearest integer if given a float", () => {
+        return request(app)
+            .patch("/api/articles/1")
+            .send({ inc_votes: 1.99997 })
+            .expect(200)
+            .then(({ body: { article } }) => {
+                expect(article.votes).toBe(101);
+            });
+    });
+    test("PATCH - 400: returns error when passed obj doesn't have an inc_votes property", () => {
+        return request(app)
+            .patch("/api/articles/1")
+            .send({ irrelevant_property: "something irrelevant" })
```

Comment:
Be careful of controlling your desired changes here. This features both a non-expected key and a string as the value rather than a number. Extracting what is causing the error should be down to a single change in testing.

Reply:
Fair point. I shall make it into an integer